libs/maps/include/mrpt/maps/CPointsMap.h

```diff
@@ -1200,6 +1200,10 @@ class PointCloudAdapter<mrpt::maps::CPointsMap>
 	inline size_t size() const { return m_obj.size(); }
 	/** Set number of points (to uninitialized values) */
 	inline void resize(const size_t N) { m_obj.resize(N); }
+	/** Does nothing as of now */
+	inline void setDimensions(const size_t& height, const size_t& width)
```

Comment:
I think this method does NOT need to be added here... it's enough to have the empty setDimensions() method in the PointCloudAdapter specialization for all MRPT classes, and only make it to have a real effect in the PCL specialization, right?

Reply:
I had to add the method to other specializations too because `project3DPointsFromDepthImageInto` gets instantiated using the non-pcl types too. For instance, [here](https://github.com/MRPT/mrpt/blob/970cf8b280b3b8e767292013aeeb01c4485dce5b/libs/obs/src/CObservation3DRangeScan.cpp#L1089).

Reply:
> because project3DPointsFromDepthImageInto gets instantiated using the non-pcl types too.

Sure... but if I recall it right, it internally instantiates a PointCloudAdapter (PCA) of the required target type as template argument. Then, my point is: shouldn't it be `setDimensions()` a method of the PCA instead of the base, target types? In this way, only one PCA should be modified to add a dummy (empty)  `setDimensions()` method, the version for all MRPT classes. Only the specialization for PCL should have a `setDimensions()` method with real stuff inside.

Let me know if this makes now sense to you...

Reply:
Yes, that makes sense. I did try to do that at first, but I couldn't actually find the definition for the class template `PointCloudAdapter` adapter anywhere... The class declaration is [here](https://github.com/MRPT/mrpt/blob/deca6243a8ddd4b36ac450f1c4f9316a87f3fa0e/libs/opengl/include/mrpt/opengl/pointcloud_adapters.h#L38), but only specialization specific definitions have been written, I think.

Or maybe I'm wrong..

Reply:
Update: my fault. I didn't realize you added the methods to `PointCloudAdapter<mrpt::maps::CPointsMap>`, it seemed to me that the method was added to `CPointsMap` directly! So, it's good!